Devscord.DiscordFramework/Middlewares/Factories/ChannelContextFactory.cs

```diff
@@ -1,14 +1,15 @@
 ﻿using Devscord.DiscordFramework.Middlewares.Contexts;
+using Discord;
 using Discord.Rest;
 using Discord.WebSocket;
 
 namespace Devscord.DiscordFramework.Middlewares.Factories
 {
-    internal class ChannelContextFactory : IContextFactory<IRestMessageChannel, ChannelContext>
+    internal class ChannelContextFactory : IContextFactory<IMessageChannel, ChannelContext>
     {
-        public ChannelContext Create(IRestMessageChannel restChannel)
+        public ChannelContext Create(IMessageChannel messageChannel)
         {
-            return restChannel == null ? null : new ChannelContext(restChannel.Id, restChannel.Name);
+            return messageChannel == null ? null : new ChannelContext(messageChannel.Id, messageChannel.Name);
```

Comment:
jak tutaj ktoś wrzuci nulla to powinno wywalić wyjątek
nigdy nie powinno być takiej sytuacji

Reply:
Jakby co to ten mechanizm ktoś inny wymyślił, ale mogę go zmienić tak jak ty proponujesz.

Reply:
czyli zostawić ustawianie tego jako nulla czy wyrzucać wyjątek?

Reply:
myśle że lepiej sprawdzać wyżej